sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -585,4 +596,130 @@ TEST_F(KmacEndTest, Error) {
   op_state_.squeezing = false;
   EXPECT_EQ(dif_kmac_end(&kmac_, &op_state_), kDifError);
 }
+
+class KmacConfigureTest : public KmacTest {
+ protected:
+  dif_kmac_config_t kmacConfig_ = {
+      .entropy_mode = kDifKmacEntropyModeIdle,
+      .entropy_fast_process = false,
+      .entropy_seed = 0x45987a984f9f862d,
+      .entropy_reseed_interval = 4985,
+      .entropy_wait_timer = 9562,
+      .message_big_endian = false,
+      .output_big_endian = false,
+      .sideload = false,
+      .msg_mask = true,
+  };
+  KmacConfigureTest() {
+    config_reg_.msg_big_endian = kmacConfig_.message_big_endian;
+    config_reg_.state_big_endian = kmacConfig_.output_big_endian;
+    config_reg_.entropy_mode = kmacConfig_.entropy_mode;
+    config_reg_.entropy_fast_process = kmacConfig_.entropy_fast_process;
+    config_reg_.sideload = kmacConfig_.sideload;
+    config_reg_.key_strength = 0;
+    config_reg_.mode = 0;
+    config_reg_.msg_mask = kmacConfig_.msg_mask;
+  }
+};
+
+TEST_F(KmacConfigureTest, Success) {
+  EXPECT_READ32(KMAC_STATUS_REG_OFFSET, {{KMAC_STATUS_SHA3_IDLE_BIT, true}});
+  ExpectConfig();
+  EXPECT_WRITE32(KMAC_ENTROPY_PERIOD_REG_OFFSET,
+                 {{KMAC_ENTROPY_PERIOD_WAIT_TIMER_OFFSET,
+                   kmacConfig_.entropy_wait_timer}});
+  ExpectEntropySeed(kmacConfig_.entropy_seed);
+  EXPECT_DIF_OK(dif_kmac_configure(&kmac_, kmacConfig_));
+}
+
+TEST_F(KmacConfigureTest, BadArg) {
+  EXPECT_DIF_BADARG(dif_kmac_configure(NULL, kmacConfig_));
+  kmacConfig_.entropy_mode = (dif_kmac_entropy_mode_t)0xff;
+  EXPECT_DIF_BADARG(dif_kmac_configure(&kmac_, kmacConfig_));
+}
+
+TEST_F(KmacConfigureTest, Locked) {
+  EXPECT_READ32(KMAC_STATUS_REG_OFFSET, {{KMAC_STATUS_SHA3_IDLE_BIT, false}});
+  EXPECT_EQ(dif_kmac_configure(&kmac_, kmacConfig_), kDifLocked);
+}
+
+class KmacStatusTest : public KmacTest {
+ protected:
+  dif_kmac_status_t status_;
+  size_t fifo_depth = KMAC_STATUS_FIFO_DEPTH_MASK;
+};
+
+TEST_F(KmacStatusTest, IdleFifoEmptySuccess) {
+  fifo_depth = 0;
```

Comment:
Does it really make sense to have this variable that you indirect through? I would just write the value directly in the one or two places it appears per test.

Reply:
Done